packages/discord.js/src/managers/GuildManager.js

```diff
@@ -164,81 +164,80 @@ class GuildManager extends CachedManager {
    */
   async create({
     name,
-    afkChannelId,
-    afkTimeout,
-    channels = [],
+    icon = null,
+    verificationLevel,
     defaultMessageNotifications,
     explicitContentFilter,
-    icon = null,
     roles = [],
+    channels = [],
+    afkChannelId,
+    afkTimeout,
     systemChannelId,
     systemChannelFlags,
-    verificationLevel,
   }) {
-    icon = await DataResolver.resolveImage(icon);
-
-    for (const channel of channels) {
-      channel.parent_id = channel.parentId;
-      delete channel.parentId;
-      channel.user_limit = channel.userLimit;
-      delete channel.userLimit;
-      channel.rate_limit_per_user = channel.rateLimitPerUser;
-      delete channel.rateLimitPerUser;
-      channel.rtc_region = channel.rtcRegion;
-      delete channel.rtcRegion;
-      channel.video_quality_mode = channel.videoQualityMode;
-      delete channel.videoQualityMode;
-
-      if (!channel.permissionOverwrites) continue;
-      for (const overwrite of channel.permissionOverwrites) {
-        overwrite.allow &&= PermissionsBitField.resolve(overwrite.allow).toString();
-        overwrite.deny &&= PermissionsBitField.resolve(overwrite.deny).toString();
-      }
-      channel.permission_overwrites = channel.permissionOverwrites;
-      delete channel.permissionOverwrites;
-    }
-    for (const role of roles) {
-      role.color &&= resolveColor(role.color);
-      role.permissions &&= PermissionsBitField.resolve(role.permissions).toString();
-    }
-    systemChannelFlags &&= SystemChannelFlagsBitField.resolve(systemChannelFlags);
-
     const data = await this.client.rest.post(Routes.guilds(), {
       body: {
         name,
-        icon,
+        icon: icon && (await DataResolver.resolveImage(icon)),
         verification_level: verificationLevel,
         default_message_notifications: defaultMessageNotifications,
         explicit_content_filter: explicitContentFilter,
-        roles,
-        channels,
+        roles: roles.map(({ color, permissions, ...options }) => ({
+          ...options,
+          color: color && resolveColor(color),
+          permissions: permissions && PermissionsBitField.resolve(permissions).toString(),
+        })),
+        channels: channels.map(
+          ({
+            parentId,
+            userLimit,
+            rtcRegion,
+            videoQualityMode,
+            permissionOverwrites,
+            rateLimitPerUser,
+            ...options
+          }) => ({
+            ...options,
+            parent_id: parentId,
+            user_limit: userLimit,
+            rtc_region: rtcRegion,
+            video_quality_mode: videoQualityMode,
+            permission_overwrites: permissionOverwrites?.map(({ allow, deny, ...options2 }) => ({
+              ...options2,
```

Comment:
This doesn't read the outer scope's `options`, you can shadow the variable by renaming `options2` to `options`.

Reply:
Couldn't do this due to ESLint no-shadow, so I renamed it to `permissionOverwriteOptions` instead.